packages/stryker/src/initializer/StrykerInitializer.ts

```diff
@@ -0,0 +1,133 @@
+      selectedTestFramework,
+      selectedReporters,
+      npmDependencies,
+      await this.fetchAdditionalConfig(npmDependencies)).write();
+    this.out('Done configuring stryker. Please review `stryker.conf.js`, you might need to configure your files and test runner correctly.');
+    this.out('Let\'s kill some mutants with this command: `stryker run`');
+  }
+
+  /**
+  * The typed rest client works only with the specific HTTP_PROXY and HTTPS_PROXY env settings.
+  * Let's make sure they are available.
+  */
+  private patchProxies() {
+    const copy = (from: string, to: string) => {
+      if (process.env[from] && !process.env[to]) {
+        process.env[to] = process.env[from];
+      }
+    };
+    copy('http_proxy', 'HTTP_PROXY');
+    copy('https_proxy', 'HTTPS_PROXY');
+  }
+
+  private async selectTestRunner(): Promise<PromptOption | null> {
+    const testRunnerOptions = await this.client.getTestRunnerOptions();
+    if (testRunnerOptions.length) {
+      const testRunnerOptions = await this.client.getTestRunnerOptions();
+      log.debug(`Found test runners: ${JSON.stringify(testRunnerOptions)}`);
+      return await this.inquirer.promptTestRunners(testRunnerOptions);
+    } else {
+      this.out('Unable to select a test runner. You will need to configure it manually.');
+      return null;
+    }
+  }
+
+  private async selectReporters(): Promise<PromptOption[]> {
+    let reporterOptions: PromptOption[];
+    try {
+      reporterOptions = await this.client.getTestReporterOptions();
+    } catch (err) {
+      this.out('Unable to fetch additional reporters.');
+      reporterOptions = [];
+    }
+    reporterOptions.push({
+      name: 'clear-text',
+      npm: null
+    }, {
+        name: 'progress',
+        npm: null
+      });
+    return this.inquirer.promptReporters(reporterOptions);
+  }
+
+  private async selectTestFramework(testRunnerOption: PromptOption): Promise<null | PromptOption> {
+    let selectedTestFramework: PromptOption | null = null;
+    const testFrameworkOptions = await this.client.getTestFrameworkOptions(testRunnerOption.npm);
+    if (testFrameworkOptions.length) {
+      log.debug(`Found test frameworks for ${testRunnerOption.name}: ${JSON.stringify(testFrameworkOptions)}`);
+      const none = {
+        name: 'None/other',
+        npm: null
+      };
+      testFrameworkOptions.push(none);
+      selectedTestFramework = await this.inquirer.promptTestFrameworks(testFrameworkOptions);
+      if (selectedTestFramework === none) {
+        selectedTestFramework = null;
+        this.out('OK, downgrading coverageAnalysis to "all"');
+      }
+    } else {
+      this.out(`No stryker test framework plugin found that is compatible with ${testRunnerOption.name}, downgrading coverageAnalysis to "all"`);
+    }
+    return selectedTestFramework;
+  }
+
+  private getSelectedNpmDependencies(testRunner: PromptOption | null, testFramework: PromptOption | null, reporters: PromptOption[]) {
```

Comment:
Why do we have a parameter for each option? Why not `getSelectedNpmDependencies(dependencies: PomptOption[])`? That could simplify this method. If we pass in one parameter, we could also call if from `installNpmDependencies`